src/frontend/planner_test/tests/testdata/share.yaml

```diff
@@ -0,0 +1,123 @@
+        CountBids.starttime_c
+      FROM (
+        SELECT
+          count(*) AS num,
+          window_start AS starttime_c
+        FROM HOP(bid, date_time, INTERVAL '2' SECOND, INTERVAL '10' SECOND)
+        GROUP BY
+          bid.auction,
+          window_start
+      ) AS CountBids
+      GROUP BY
+        CountBids.starttime_c
+    ) AS MaxBids
+    ON AuctionBids.starttime = MaxBids.starttime_c AND AuctionBids.num >= MaxBids.maxn;
+  batch_plan: |
+    BatchExchange { order: [], dist: Single }
+    └─BatchProject { exprs: [auction, count] }
+      └─BatchFilter { predicate: (count >= max(count)) }
+        └─BatchHashJoin { type: Inner, predicate: window_start = window_start, output: all }
+          ├─BatchExchange { order: [], dist: HashShard(window_start) }
+          | └─BatchProject { exprs: [auction, count, window_start] }
+          |   └─BatchHashAgg { group_key: [window_start, auction], aggs: [count] }
+          |     └─BatchHopWindow { time_col: date_time, slide: 00:00:02, size: 00:00:10, output: [auction, window_start] }
+          |       └─BatchExchange { order: [], dist: HashShard(auction) }
+          |         └─BatchProject { exprs: [auction, date_time] }
+          |           └─BatchSource { source: "bid", columns: ["auction", "bidder", "price", "date_time", "_row_id"] }
+          └─BatchProject { exprs: [max(count), window_start] }
+            └─BatchHashAgg { group_key: [window_start], aggs: [max(count)] }
+              └─BatchExchange { order: [], dist: HashShard(window_start) }
+                └─BatchHashAgg { group_key: [auction, window_start], aggs: [count] }
+                  └─BatchHopWindow { time_col: date_time, slide: 00:00:02, size: 00:00:10, output: [auction, window_start] }
+                    └─BatchExchange { order: [], dist: HashShard(auction) }
+                      └─BatchProject { exprs: [auction, date_time] }
+                        └─BatchSource { source: "bid", columns: ["auction", "bidder", "price", "date_time", "_row_id"] }
+  stream_plan: |
+    StreamMaterialize { columns: [auction, num, window_start(hidden), window_start#1(hidden)], pk_columns: [window_start, auction, window_start#1] }
+    └─StreamProject { exprs: [auction, count, window_start, window_start] }
+      └─StreamFilter { predicate: (count >= max(count)) }
+        └─StreamHashJoin { type: Inner, predicate: window_start = window_start, output: all }
+          ├─StreamExchange { dist: HashShard(window_start) }
+          | └─StreamProject { exprs: [auction, count, window_start] }
+          |   └─StreamAppendOnlyHashAgg { group_key: [window_start, auction], aggs: [count, count] }
+          |     └─StreamExchange { dist: HashShard(auction, window_start) }
+          |       └─StreamHopWindow { time_col: date_time, slide: 00:00:02, size: 00:00:10, output: [auction, window_start, _row_id] }
+          |         └─StreamProject { exprs: [auction, date_time, _row_id] }
+          |           └─StreamShare { id = 585 }
```

Comment:
So `Project` cannot be pushed down yet?

Reply:
Currently, we put the share source logic between the end of the logical optimization and the beginning of the stream optimization. So no other optimizations will be performed for the shared source. I think we will put this shared source logic together with the share cte and share view logic to the binder/planner phase in the future.

Reply:
LGTM